src/AutorestSwift/Code Generator/Manager.swift

```diff
@@ -81,7 +81,11 @@ class Manager {
     private static func sanitize(yaml: String) -> String {
         // TODO: Remove when issue (https://github.com/Azure/autorest.swift/issues/47) is fixed.
         // Replaces empty string with a single space.
-        return yaml.replacingOccurrences(of: ": ''", with: ": ' '")
+        // Keep the empty string for Value. Replace all empty string for value with PreserveEmptyString. After globally replace
+        // '' with ' ', replace PreserveEmptyString with ''
+        let preserveEmptyValue = yaml.replacingOccurrences(of: "value: ''", with: "value: PreserveEmptyString")
+        let removedGlobalEmpty = preserveEmptyValue.replacingOccurrences(of: ": ''", with: ": ' '")
+        return removedGlobalEmpty.replacingOccurrences(of: "value: PreserveEmptyString", with: "value: ''")
```

Comment:
What is the rationale for this?

Reply:
In yaml file of url.json, it has ConstantSchema with '' as value:

```
  - !<!ConstantSchema> &ref_27                                                                                        
      type: constant                                                                                                    
      value: !<!ConstantValue>                                                                                          
        value: ''                                                                                                       
      valueType: *ref_0                                                                                                 
      language: !<!Languages>                                                                                           
        default:                                                                                                        
          name: Constant14                                                                                              
          description: ''             
```  
And this constantSchema is used as path/query parameter in one of the generated method.
But if we globally replacing all '' to ' ' in the yaml file, the value of this constant will changed incorrectly by ^ workaround. As a result, we build the path/query incorrectly in the generated code. So we need to skip changing '' to ' ' in `value:`

Reply:
Revert the change. Open this issue to track for a better solution for the yams issue. https://github.com/Azure/autorest.swift/issues/214